test/test_tensor_creation_ops.py

```diff
@@ -1255,6 +1255,14 @@ def test_randperm(self, device):
             torch.randperm(n, out=non_contiguous_tensor)
             self.assertEqual(non_contiguous_tensor, res)
 
+        # Test exceptions when device and generator types are incompatible
```

Comment:
using a separate definition of test call
```
def test_device_compatibility():
   # ...
```
might be a better idea. slow tests are not actually run on every configuration.